cmd/server.go

```diff
@@ -270,52 +275,192 @@ func serveMux(ctx context.Context) (*http.ServeMux, error) {
+		return fmt.Errorf("no such field: %s in struct", fieldName)
+	}
+
+	if !field.CanSet() {
+		return fmt.Errorf("cannot set field: %s", fieldName)
+	}
+
+	fieldValue := reflect.ValueOf(value)
+	if field.Type() != fieldValue.Type() {
+		return fmt.Errorf("provided value type doesn't match field type")
 	}
 
-	login := &discovery.LoginV1{
-		Client:     flagAuthOidcClientId,
-		GrantTypes: flagLoginGrantTypes,
-		Authz:      provider.AuthURL(),
-		Token:      provider.TokenURL(),
-		Ports:      flagLoginPorts,
-		Scopes:     flagAuthOidcScopes,
+	field.Set(fieldValue)
+	return nil
+}
+
+
+func parseOidc(ctx context.Context) ([]auth.OidcConfig, error) {
+    parsedList := []auth.OidcConfig{}
+
+    if len(flagAuthOidc) != 0 {
+        fmt.Printf("flagAuthOidc: %v\n", flagAuthOidc)
+        for _, oidcConfig := range flagAuthOidc {
+            parsed := &auth.OidcConfig{
+                ClientID:    "",
+                Issuer:      "",
+                Scopes:      []string{},
+                LoginGrants: flagLoginGrantTypes,
+                LoginPorts:  flagLoginPorts,
+            }
+
+            pairs := strings.Split(oidcConfig, ";")
+
+            for _, pair := range pairs {
+                if pair == "" {
+                    continue
+                }
+                kv := strings.SplitN(pair, "=", 2)
+                if len(kv) != 2 {
+                    return nil, fmt.Errorf("invalid key-value pair: %s", pair)
+                }
+                key := strings.TrimSpace(kv[0])
+                value := strings.TrimSpace(kv[1])
+
+                if key == "scopes" || key == "login_grants" {
+                    err := setFieldByKey(parsed, key, strings.Split(value, ","))
+                    if err != nil {
+                        return nil, fmt.Errorf("invalid OIDC configuration %s: %w", key, err)
+                    }
+                } else if key == "login_ports" {
+                    ports := strings.Split(value, ",")
+                    intPorts := []int{}
+                    for _, port := range ports {
+                        intPort, err := strconv.Atoi(strings.TrimSpace(port))
+                        if err != nil {
+                            return nil, fmt.Errorf("invalid port value: %s", port)
+                        }
+                        intPorts = append(intPorts, intPort)
+                    }
+                    err := setFieldByKey(parsed, key, intPorts)
+                    if err != nil {
+                        return nil, fmt.Errorf("invalid OIDC configuration %s: %w", key, err)
+                    }
+                } else {
+                    err := setFieldByKey(parsed, key, value)
+                    if err != nil {
+                        return nil, fmt.Errorf("invalid OIDC configuration %s: %w", key, err)
+                    }
+                }
+            }
+            parsedList = append(parsedList, *parsed)
+        }
+	} else {
+		slog.Debug("setting up oidc auth",
```

Comment:
The identation isn't correct here and in a few other places. You are mixing tabs and spaces.